src/main/java/net/covers1624/quack/collection/ColUtils.java

```diff
@@ -239,12 +251,7 @@ public static <T> T headOrDefault(Iterable<T> col, @Nullable T _default) {
      * @return The optional result.
      */
     public static <T> Optional<T> tailOption(Iterable<T> col) {
-        Iterator<T> itr = col.iterator();
-        T last = null;
-        while (itr.hasNext()) {
-            last = itr.next();
-        }
-        return Optional.ofNullable(last);
+        return Optional.ofNullable(tailOrDefault(col));
```

Comment:
We can do better here. There's no reason why we can't support `Iterable<@Nullable T>`

Just copy the impl of `tailOrDefault`